ProcessLib/TimeLoop.cpp

```diff
@@ -405,6 +405,14 @@ void TimeLoop::initialize()
     // init solution storage
     _process_solutions = setInitialConditions(_start_time, _per_process_data);
 
+    if (_chemical_system != nullptr)
+    {
+        BaseLib::RunTime time_phreeqc;
+        time_phreeqc.start();
+        _chemical_system->executeInitialCalculation(_process_solutions);
```

Comment:
Just an idea; Can this initialization happen in the `ComponentTransport::setInitialConditions`?

Reply:
Would be nice. Need some time to dig deeper.

Reply:
For instance, if you like to transfer the `ChemistryLib` to another process, wouldn't it be more general to have to initialization as @renchao-lu implemented?
When calling from `ComponentTransport::setInitialConditions` you would have to define the function for other processes (`Richards Flow`etc) seperately, right?

Reply:
Maybe, important to know 1) how much additional computational time is needed if an additional equilibrium step is included and 2) if this additional step give more accurate results or not.

Reply:
The step added by @renchao-lu  will just be called once during simulation start-up. It will not substantiall increase the overall computational cost.

Reply:
Sorry for the late reply. I just come back. As Johannes pointed out, the added equilibrium calculation is executed once before time looping.